transport/tchannel/tracing.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2024 Uber Technologies, Inc.
```

Comment:
I should have made it more clear - let's put it under `transport/tchannel/tracing/tracing.go`, otherwise it is still under the same Go package that will cause circular dependency later.